### PR TITLE
Implement `send` with new protocol buffers

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Implement `send` for the  `XpringClient` which uses official rippled protocol buffers. 

Add unit and integration tests. The unit tests are taken from the `LegacyDefaultXpringClient` tests, which ensure that the implementation behaves consistently across protocol buffers. 

### Context of Change

rippled implemented protocol buffer support natively, which uses a new and incompatible set of protocol buffers. The implementation in rippled was completed in: https://github.com/ripple/rippled/pull/3159.

Protocol buffer upgrades are being done in piecemeal fashion for ease of review. 

*Status*

- [x] [Generate new versions of the protocol buffers](https://github.com/xpring-eng/Xpring-JS/pull/108)
- [x] [Migrate existing code to a `legacy` subdirectory and namespace](https://github.com/xpring-eng/Xpring-JS/pull/109)
- [x] Create code that works with the new protocol buffers from the rippled team
   - [x] [Create a new `NetworkClient` with fakes](https://github.com/xpring-eng/Xpring-JS/pull/111)
   - [x] [Create a new `DefaultXpringClient` with tests](https://github.com/xpring-eng/Xpring-JS/pull/114/files)
   - [x] Read only calls for new `DefaultXpringClient` (many PRs)
   - [x] Write calls fro `DefaultXpringClient`
- [x] [Add a boolean parameter to switch between implementations](https://github.com/xpring-eng/Xpring-JS/pull/130)

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After

Users who use the new protocol buffers will be able to send XRP.
## Test Plan

Unit tests on CI continue to pass.  Integration test passes.

### CODE REVIEW

File: src/default-xpring-client.ts [hunk 1]
```diff
@@ -159,11 +170,70 @@ class DefaultXpringClient implements XpringClientDecorator {
...
-    _sender: Wallet,
+    amount: BigInteger | number | string,
+    destination: string,
+    sender: Wallet,
   ): Promise<string> {
-    throw new Error(XpringClientErrorMessages.unimplemented)
+    if (!Utils.isValidXAddress(destination)) {
+      return Promise.reject(
+        new Error(XpringClientErrorMessages.xAddressRequired),
+      )
```
Comment (Stormtv) [inline src/default-xpring-client.ts]: throw is same as return Promise.reject

File: src/default-xpring-client.ts [hunk 2]
```diff
@@ -159,11 +170,70 @@ class DefaultXpringClient implements XpringClientDecorator {
...
+    }
+
+    const classicAddress = Utils.decodeXAddress(sender.getAddress())
+    if (!classicAddress) {
+      return Promise.reject(
+        new Error(XpringClientErrorMessages.xAddressRequired),
+      )
```
Comment (Stormtv) [inline src/default-xpring-client.ts]: Same

File: src/default-xpring-client.ts [hunk 3]
```diff
@@ -159,11 +170,70 @@ class DefaultXpringClient implements XpringClientDecorator {
...
+    const account = new AccountAddress()
+    account.setAddress(sender.getAddress())
+
+    const transaction = new Transaction()
+    transaction.setAccount(account)
+    transaction.setFee(fee)
+    transaction.setSequence(accountData.getSequence())
+    transaction.setPayment(payment)
+    transaction.setLastLedgerSequence(
+      lastValidatedLedgerSequence + ledgerSequenceMargin,
```
Comment (Stormtv) [inline src/default-xpring-client.ts]: So do we need to use `lastValidatedLedgerSequence.getIndex()` here instead? This is similar to that balance thing we were talking about yesterday.

Comment (keefertaylor) [reply src/default-xpring-client.ts]: `getLastValidatedLedgerSequence()` returns a `number` already, so this is a `number` + `number`.

Comment (Stormtv) [reply src/default-xpring-client.ts]: My bad on this one, the old legacy protobuffs return an object.

File: src/default-xpring-client.ts [hunk 4]
```diff
@@ -205,6 +271,43 @@ class DefaultXpringClient implements XpringClientDecorator {
...
+   */
+  private static toBigInt(value: string | number | BigInteger): BigInteger {
+    if (typeof value === 'string') {
+      return bigInt(value)
+    }
+    if (typeof value === 'number') {
+      return bigInt(value)
+    }
+    // Value is already a BigInteger.
+    return value
```
Comment (intelliot) [inline src/default-xpring-client.ts]: There should be no harm in returning `bigInt(someBigInteger)` so I think this entire function can be replaced with just `return bigInt(value)`

File: src/default-xpring-client.ts [hunk 5]
```diff
@@ -8,20 +8,31 @@ import GRPCNetworkClientWeb from './grpc-network-client.web'
...
+import { Payment, Transaction } from './generated/web/rpc/v1/transaction_pb'
+import { AccountRoot } from './generated/web/rpc/v1/ledger_objects_pb'
 
 // TODO(keefertaylor): Re-enable this rule when this class is fully implemented.
 /* eslint-disable @typescript-eslint/require-await */
 /* eslint-disable @typescript-eslint/no-unused-vars */
 /* eslint-disable class-methods-use-this */
 
+/** A margin to pad the current ledger sequence with when submitting transactions. */
+const ledgerSequenceMargin = 10
```
Comment (intelliot) [inline src/default-xpring-client.ts]: I recommend renaming `ledgerSequenceMargin` to `maxLedgerVersionOffset` for consistency with existing APIs, unless we have a good reason to introduce new terminology. https://xrpl.org/rippleapi-reference.html#transaction-instructions

Comment (keefertaylor) [reply src/default-xpring-client.ts]: Done. Thanks for the tip.

File: src/default-xpring-client.ts (hunk 4 above)
Comment (keefertaylor) [reply src/default-xpring-client.ts]: This actually throws a bunch of typing errors (it wants a unary value, not a complex value type).

Comment (keefertaylor) [reply src/default-xpring-client.ts]: Oh, this does work. Weirdly, VSCode complains but it compiles!

**Pull request merged**